src/reducers/Student.jsx

```diff
@@ -0,0 +1,163 @@
+import { combineReducers } from 'redux';
```

Comment:
Overall I think that the new state shape is much nicer. If the stuff that you've tried appears to work, I say we ship this. My one significant thought is that it seems like the reducers which are dependent on database calls should probably have cases to handle when the database request fails or times out. But the "nothing" behavior is a sane default, so we're probably good.